### PR TITLE
cli: use StringArray over StringSlice Part 1

### PR DESCRIPTION
**Pull request by Luap99**
see commits

This is only part 1 as there are much more to go but I wanted to keep the review size reasonable.

#### Does this PR introduce a user-facing change?

```release-note
The following cli options have been changed so that multiple values are no longer split by comma. This allows values with comma in them to be used:
podman --module
podman --hooks-dir
podman run/create --annotation
podman run/create --env-file
podman run/create --log-opt
podman run/create --chrootdirs
podman run/create --decryption-key
podman run/create --label-file
podman run/create --device
podman run/create --device-{read,write}-iops
podman run/create --device-{read,write}-bps
podman run/create --blkio-weight-device
podman exec --env-file
```

### CODE REVIEW

**Comment by Luap99** [comment]
@edsantiago PTAL at the test changes, I mostly went with the easy option to adapt an existing test but some options had 0 tests so I had to create some new ones.

File: cmd/podman/README.md [hunk 1]
```diff
@@ -109,3 +109,11 @@ The complete set can be found in the `validate` package, here are some examples:
     created := validate.ChoiceValue(&opts.Sort, "command", "created", "id", "image", "names", "runningfor", "size", "status")
     flags.Var(created, "sort", "Sort output by: "+created.Choices())
     ```
+
+## Adding CLI flags
+
+When adding adding anew cli option that accepts a string array then there are two options to choose from: `StringSlice()` and `StringArray()`.
```
Comment (edsantiago) [inline cmd/podman/README.md:115]: - Missing space: "a new", not anew
- suggestion, change "a string array then there are" -> "a string array, there are"

Apologies in advance for possible email bombardment. This is a big PR, and I've been bitten before by github trashing review comments when someone repushes, so I am going to submit feedback one by one. There is no need to repush for every comment. I will give an "all done" signal when I've finished my first pass.

File: cmd/podman/README.md [hunk 2]
```diff
@@ -109,3 +109,11 @@ The complete set can be found in the `validate` package, here are some examples:
...
+They differ slightly in their behavior, `StringSlice()` allows the values to be comma separated so `--opt v1,v2 --opt v3` results in
```
Comment (edsantiago) [inline cmd/podman/README.md:116]: suggestion: comma -> colon in "...in their behavior: ..."

File: test/system/300-cli-parsing.bats [hunk 3]
```diff
@@ -158,6 +158,18 @@ EOF
     expect[weird*na#me!]=$weirdname
 
     _check_env $resultsfile
+
+    # Now check the same with podman exec
+    run_podman run -d --name testctr        \
+            -v "$resultsfile:/envresults:Z" \
+            $IMAGE top
+
```
Comment (edsantiago) [inline test/system/300-cli-parsing.bats:166]: Is `wait_for_output "Load average:"` needed here, to guarantee that the volume is mounted and the container is ready for an `exec`? (I think not, i.e. I think what you're doing is safe, but would like explicit confirmation).

**Review by edsantiago — COMMENTED** [review]
All done! Except I think there are still a number of `StringSliceVar()`s that can be converted, but I'm sure you're aware of those. [EDIT: yes, I just saw your github PR comment. I had only been reading git commit messages.]

This is really nice work. Thank you for catching all these and for adding clean tests.

File: test/system/300-cli-parsing.bats (hunk 3 above)
Comment (Luap99) [reply test/system/300-cli-parsing.bats:166]: It should not be needed here, podman run -d waits for the process to be started before returning which means all mounts etc are setup already. It is only needed the test depends on the container process to do something.

**Comment by Luap99** [comment]
@edsantiago Thanks for the comments, should be all resolved now if I didn't mess up the interactive rebase.

**Comment by edsantiago** [comment]
LGTM but I think you need to edit push/pull.go -- or remove the decryptionkey test change for now.

**Comment by Luap99** [comment]
I dropped the test for now, will add it back once I get to the other flags.

**Comment by Luap99** [comment]
@containers/podman-maintainers PTAL

**Comment by Luap99** [comment]
> LGTM. Wish we had a linter for this.

We can write stuff if we think it is important enough, the main issue is that there are valid use cases for StringSlice so how should a linter make this choice? Or do you want the linter to warn about every use of StringSlice and require legit users to use `//nolint`?

**Comment by rhatdan** [comment]
/lgtm
I too vote for a linter and //nolint solution.

**Pull request merged**